src/mesh/config/MeshConfiguration.cpp

```diff
@@ -52,30 +58,25 @@ MeshConfiguration::MeshConfiguration(
   parent.addSubtag(tag);
 }
 
-void MeshConfiguration::setDimensions(
-    int dimensions)
-{
-  PRECICE_TRACE(dimensions);
-  PRECICE_ASSERT((dimensions == 2) || (dimensions == 3), dimensions);
-  _dimensions = dimensions;
-}
-
 void MeshConfiguration::xmlTagCallback(
     const xml::ConfigurationContext &context,
     xml::XMLTag &                    tag)
 {
   PRECICE_TRACE(tag.getName());
   if (tag.getName() == TAG) {
-    PRECICE_ASSERT(_dimensions != 0);
-    std::string name = tag.getStringAttributeValue(ATTR_NAME);
+    std::string name       = tag.getStringAttributeValue(ATTR_NAME);
+    int         dimensions = tag.getIntAttributeValue(ATTR_DIMENSIONS);
+    _meshDimensionsMap.insert(std::pair<std::string, int>(name, dimensions));
+    PRECICE_ASSERT(dimensions != 0);
     PRECICE_ASSERT(_meshIdManager);
-    _meshes.push_back(std::make_shared<Mesh>(name, _dimensions, _meshIdManager->getFreeID()));
+    _meshes.push_back(std::make_shared<Mesh>(name, dimensions, _meshIdManager->getFreeID()));
   } else if (tag.getName() == TAG_DATA) {
     std::string name  = tag.getStringAttributeValue(ATTR_NAME);
     bool        found = false;
     for (const DataConfiguration::ConfiguredData &data : _dataConfig->data()) {
+      auto dataDimensions = getDataDimensions(_meshes.back()->getName(), data.typeName);
```

Comment:
We need to access the mesh name somehow, this was the way we tried, it seems to work.